core/src/main/scala/tech/beshu/ror/boot/ReadonlyRest.scala

```diff
@@ -89,25 +101,61 @@ class ReadonlyRest(coreFactory: CoreFactory,
     }
   }
 
+  private def runTestStartingFailureProgram[A](action: TestConfigLoading.LoadTestRorConfig[TestConfigLoading.ErrorOr[A]]): EitherT[Task, StartingFailure, A] = {
+    val compiler = TestConfigLoadingInterpreter.create(indexTestConfigManager, RorProperties.rorIndexSettingLoadingDelay)
+    EitherT(action.foldMap(compiler))
+      .leftMap(toStartingFailure)
+  }
+
+  private def toStartingFailure(error: LoadedTestRorConfig.Error) = {
+    error match {
+      case LoadedTestRorConfig.IndexParsingError(message) =>
+        StartingFailure(message)
+      case LoadedTestRorConfig.IndexUnknownStructure =>
+        StartingFailure(s"Test settings index is malformed")
```

Comment:
I'm not sure if we should not allow starting ROR when test settings are invalid. 

The test settings are used by an admin. It doesn't apply to users, so it should not affect them. 
Maybe we can just ignore it, write a big error in the log and continue? Do you see any problems with this approach?

Reply:
I created an appropriate fallback config for each failure case.